persistence/block.go

```diff
@@ -55,15 +76,51 @@ func (p PostgresContext) TransactionExists(transactionHash string) (bool, error)
 	return true, err
 }
 
-func (p PostgresContext) StoreTransaction(txResult modules.TxResult) error {
-	return p.txIndexer.Index(txResult)
+func (p PostgresContext) StoreTransactions() error {
+	// TODO: store in batch
+	for _, txResult := range p.GetLatestTxResults() {
+		if err := p.txIndexer.Index(txResult); err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
-func (p PostgresContext) StoreBlock(blockProtoBytes []byte) error {
+// DISCUSS: this might be retrieved from the block store - temporarily we will access it directly from the module
+//       following the pattern of the Consensus Module prior to pocket/issue-#315
+func (p *PostgresContext) SetProposalBlock(blockHash string, blockProtoBytes, proposerAddr, qc []byte, transactions [][]byte) error {
+	p.latestBlockHash = blockHash
+	p.latestBlockProtoBytes = blockProtoBytes
+	p.latestProposerAddr = proposerAddr
+	p.latestQC = qc
+	p.latestBlockTxs = transactions
+	return nil
+}
+
+// TEMPORARY: Including two functions for the SQL and KV Store as an interim solution
+//                 until we include the schema as part of the SQL Store because persistence
+//                 currently has no access to the protobuf schema which is the source of truth.
+// TODO: atomic operations needed here - inherited pattern from consensus module
+func (p PostgresContext) StoreBlock() error {
+	if p.latestBlockProtoBytes == nil {
+		// IMPROVE/CLEANUP: HACK - currently tests call Commit() on the same height and it throws a
+		// ERROR: duplicate key value violates unique constraint "block_pkey", because it attempts to
```

Comment:
I already have this done in the next PR. Code and utilities :)

Reply:
Awesome :)